src/Interpreters/join_common.cpp

```diff
@@ -276,6 +276,20 @@ void joinTotals(const Block & totals, const Block & columns_to_add, const Names
     }
 }
 
+void addDefaultValues(MutableColumnPtr & column, const DataTypePtr & typ, size_t count)
+{
+    if (typ->getTypeId() == column->getDataType())
```

Comment:
What is this branch for?

Reply:
It is aimed to use `column->insertManyDefaults` as earlier if possible to not decrease performance. But if I'll rewrite it with `IDataType::insertDefaultInto` I think this `if` can be omitted.